openapi/platform/platform.yaml

```diff
@@ -3887,26 +3890,6 @@ paths:
           in: query
           name: 'filter[recipientOrganizationId]'
           description: The id of the recipient organization
-        - schema:
-            type: array
-            enum:
-              - content
-              - lineItems
-              - currencyCode
-              - createdAt
-              - updatedAt
-              - archivedAt
-              - status
-              - viewCount
-              - shareType
-              - pdfUrl
-              - previewImageUrl
-              - title
-            example: 'title,content,status'
-            default: 'title,status,updatedAt'
-          in: query
-          name: 'fields[proposals]'
-          description: 'A comma separated list of the attributes that you want included in the response. By default the `content` is not included as it could be very large. '
```

Comment:
Not sure why we added this but we probably don't need to support this level of granularity in the first slice, if ever.

Reply:
The intention might've been for `lineItems` and `content` originally but we made the decision as a team to not allow those fields in LIST requests.

Reply:
The `fields[type]` query param is an optional part of the [JSON:API specification](https://jsonapi.org/format/#fetching-sparse-fieldsets). Support for filtering out the fields a user does not want is already [built into the marshaling to JSON](https://github.com/vendasta/api-gateway/blob/0f27191d21731764ede3132da7e84770b128f399/vendor/github.com/manyminds/api2go/api.go#L1135-L1192). 

The way I had envisioned this working is that we would add this helper to the `params` package.
```go
// GetReadFieldMask can be called within your resource handler functions to see if extra processing needs to be done to fetch complex values.
// Note the response serialization handles omitting unwanted model fields so you do not need this for basic requests
func GetReadFieldMask(r api2go.Request, resourceType model.ResourceType, defaultMask []string) (fields *fieldmask.FieldMask) {
  paramName := fmt.Sprintf("fields[%s]", resourceType)
  values, ok := r.QueryParams[paramName]
  if !ok {
    values = defaultMask
  } else if len(values) == 0 {
    // Special case for when the user explicitly told us they don't want any fields
    fm := make(fieldmask.FieldMask, 0)
    return &fm
  }
  return fieldmask.FieldMaskProtoPathsToFieldMask(values)
}
```

Then in your `FindAll` handler you would do something like 
```go
fm := params.GetReadFieldMask(r, model.BusinessLocationReviewType, []string{"title","status","updatedAt"})
if fm.IsPropertyPresent("content"){
  //Make an additional API call to get the content and add to the response object
}
if fm.IsPropertyPresent("lineItems"){
  //Make an additional API call to get the lineItems and add to the response object
}
```

Because `Update` reuses `FindOne` we can't skip fetching the updatable fields in `FindOne` without a bit more work.

With this additional context do you still feel it needs to be removed?

Reply:
I added `fields[types]` to all our `GET` and `LIST` requests but removed `content` and `lineItems` from the enum 👍🏻 

Let me know if this looks okay